src/EventStore.ClientAPI.Embedded/EmbeddedEventStorePersistentSubscription.cs

```diff
@@ -10,7 +10,7 @@ internal class EmbeddedEventStorePersistentSubscription : EventStorePersistentSu
 
         internal EmbeddedEventStorePersistentSubscription(
             string subscriptionId, string streamId,
-            Action<EventStorePersistentSubscriptionBase, ResolvedEvent> eventAppeared,
+            Func<EventStorePersistentSubscriptionBase, ResolvedEvent, Task> eventAppeared,
```

Comment:
That is absolutely an improvement I would like to see.

Unfortunately it is not required for our use case and I have no more day time to spend on this. If no one else does it, I'll try to add this in as a separate PR.